src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -156,22 +159,18 @@ object ScalafixPlugin extends AutoPlugin {
           // here, as it's a task and we are in a settings context. Therefore
           // we fallback to an empty list of rules in case of resolution error
           // to keep providing completions for the rest.
-          Try {
-            scalafixInterfaceProvider
-              .value((ThisBuild / scalafixResolvers).value)
-              .availableRules()
-          }.getOrElse(Seq.empty),
+          Try(scalafixInterfaceProvider().availableRules())
+            .getOrElse(Seq.empty),
         terminalWidth = Some(JLineAccess.terminalWidth),
         allowedTargetFilesPrefixes = allowedTargetFilesPrefixes,
         jgitCompletion = scalafixJGitCompletion.value
       )
     }(_.parser)
 
-  // Memoize ScalafixInterface instances initialized with a custom tool classpath across projects & configurations
-  // during task execution, to amortize the classloading cost when invoking scalafix concurrently on many targets
```

Comment:
local rules were relying on this cache to be cleared from one invocation to another, thus the scripted failures now that memoization persists across invocations because of the change to a setting key